launch/launch/utilities/signal_management.py

```diff
@@ -240,3 +241,107 @@ def handle(
         else:
             old_handler = self.__handlers.pop(signum, None)
         return old_handler
+
+
+
+__global_signal_manager_activated_lock = threading.Lock()
+__global_signal_manager_activated = False
+__global_signal_manager = AsyncSafeSignalManager()
+
+
+def on_sigint(handler):
+    """
+    Set the signal handler to be called on SIGINT.
+
+    Pass None for no custom handler.
+
+    install_signal_handlers() must have been called in the main thread before.
+
+    .. deprecated:: Foxy
+
+      Use AsyncSafeSignalManager instead
+    """
+    warnings.warn(
```

Comment:
I'd rather not release deprecation warnings into a long-standing ROS distro to avoid disruption. Besides, users who have already switched to Galactic would have already run into the API change without a tick-tock cycle. There may be other opinions.

Reply:
I can only give one user's opinion (mine/my team's): I'd like to get warned of breakages as soon a possible. We'd have to change it eventually in any way since we move over to newer versions of ROS2 eventually (LTS -> LTS). And if not, a `warnings.simplefilter()` ist easy to implement if one does not want to see the message for the time being.

But you know best what's common practice in ROS. This isn't Rolling Ridley, after all.

Reply:
I have to agree with @jacobperron ; I don't think we should introduce new deprecation warnings into a stable distribution.  Having new warnings show up for Foxy users as a result of `sudo apt-get update` is really user-unfriendly.

When people port to Galactic or Rolling, they'll get the warnings.

Reply:
Alright, see https://github.com/ros2/launch/pull/511/commits/d8e320a42597348ad5bbc135c58954932733a8d6. 

However I will say that, while I agree that introducing warnings into a patch release of a stable distribution is not ideal, this patch is not transparent. Signals now go through wakeup file descriptors and handling is deferred to the main-thread local `asyncio` loop (assuming it is run). Depending on how the old signal management API was used (the only case in which you'd get a warning) this can potentially break those use cases -- silently as of https://github.com/ros2/launch/pull/511/commits/d8e320a42597348ad5bbc135c58954932733a8d6.

Which is why I didn't even think of backporting anything, and I probably wouldn't if these bits were not kinda broken in Foxy.

> When people port to Galactic or Rolling, they'll get the warnings.

The old API was completely removed in Galactic, so no warnings. This patch is a retrofit of the original patch.

Reply:
I didn't fully comprehend the nature of the change. If there's a risk that we're breaking somebody, then I think issuing a warning is a good thing. I don't like the idea of releasing breaking changes in Foxy, but if you think the fix is worth the potential churn then I'd rather be very loud about it.

Reply:
So, without this fix, `launch` hangs every now and then. It's the same we've experienced with `ros2cli` tests before. And it hit @felixdivo, so it may hit others too.

I agree about being loud. I thought a warning would be OK, but perhaps it's too loud. @clalancette ?

Reply:
It seems like we are stuck between a rock and a hard place.  While I don't like introducing new warnings into a stable distribution, silently breaking users isn't good either.

In this case, I'll defer to your best judgement on what to do here.  You obviously have a much better handle on the issue and launch in general than I do.

Reply:
Presumably, the these signal APIs are not very popular so I'd hope the number of affected users would be small.

Reply:
Alright, I put deprecation warnings back in. I'll merge this PR into #506 and run CI there.